lib/honeybadger/breadcrumbs/cleaner.rb

```diff
@@ -0,0 +1,29 @@
+module Honeybadger
+  module Breadcrumbs
+    # @api private
+    # Responsible for ensuring we are storing and sending sane breadcrumb data
+    class Cleaner
+      VALID_TYPES = [TrueClass, FalseClass, Numeric, String].freeze
+
+      def initialize(config)
+        @config = config
+      end
+
+      def clean!(breadcrumb)
+        breadcrumb.metadata.keep_if(&method(:valid_metadata?))
+      end
+
+      private
+
+      def valid_metadata?(k, v)
+        return true if allowed_metadata_type?(v)
+        @config.logger.debug("Removed metadata key '#{k}' from breadcrumb because value was invalid")
+        false
+      end
+
+      def allowed_metadata_type?(value)
+        VALID_TYPES.any? { |t| value.is_a?(t) }
+      end
```

Comment:
Would it make sense to speed this up a bit by using a hash rather than a list:

`!!VALID_TYPES[value.class]`

This assumes VALID_TYPES is changed to something like this:

`Hash[[TrueClass, FalseClass, Numeric, Float, Integer].map {|c| [c, true] }]`

Reply:
Yeah, that would be faster for sure. The only thing is we would lose the inheritance checking that we get with is_a?. I'm not sure of any other types other than numbers that this would have a big effect on, so perhaps just enumerating over all the number types would be enough (as you did above)